src/components/elements/inputs/input.js

```diff
@@ -27,28 +21,36 @@ export default class Input extends Component {
     }
   }
 
-  getValue(target) {
-    return target.value
-  }
+  render() {
+    const Element = factory(this.props.property)
+    let className = 'ct-input'
 
-  onBlur() {
-    this.update(this.state.value)
-  }
+    if (this.props.property.errors.length) {
+      className += ' ct-input-invalid'
+    }
 
-  onChange(e) {
-    this.setState({
-      value: e.target.value === '' ? null : this.getValue(e.target),
-    })
+    return (
+      <Element
+        className={className}
+        property={this.props.property}
+        save={value => this.save(value)}
+        update={value => this.setState({ value })}
+        value={this.state.value} />
+    )
   }
 
-  update(value) {
-    this.setState({ value })
+  save(value) {
+    if (value === undefined) {
+      value = this.state.value
+    }
+    else {
+      this.setState({ value })
+    }
 
-    if (this.props.property.value != value) {
-      const property = this.props.property
-      const updateHref = property.links.update.href
+    const { property } = this.props
 
-      this.props.update(updateHref, property.id, value)
+    if (property.value !== value) {
+      this.props.update(property.links.update.href, property.id, value)
```

Comment:
Good point, I also found it somewhat confusing when it is called ```update``` in the api. Do you still think commit is a good name when it takes an argument?